src/components/Reservations/Reservations.js

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable max-len */
+import React, { useEffect, useState } from 'react';
+import Button from 'react-bootstrap/Button';
+import { useDispatch, useSelector } from 'react-redux';
+import { useSearchParams } from 'react-router-dom';
+import Container from 'react-bootstrap/Container';
+import Spinner from 'react-bootstrap/Spinner';
+import Pagination from 'react-bootstrap/Pagination';
+import { fechReservations } from '../../redux/reservations/reservations';
+import { deleteReservations, clearDeleteReservationState, clearFetchReservationState } from '../../redux/reservations/deleteReservation';
+import { getLawyers } from '../../redux/lawyers/lawyersIndex';
+
+const Reservations = () => {
+  const dispatch = useDispatch();
+  const reservationState = useSelector((state) => state.reservations);
+  const lawyersState = useSelector((state) => state.lawyers);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const currentPage = searchParams.get('p');
+  const [showCurrentPage, setShowCurrentPage] = useState(+currentPage || 1);
+  const [maxPagesNumber, setMaxPagesNumber] = useState(null);
+  const initialArrayIndex = ((showCurrentPage * 5) - 5);
+  const itemsToRender = reservationState.reservations.slice(initialArrayIndex, initialArrayIndex + 5);
+
+  useEffect(() => {
+    if (reservationState.status !== 'completed') {
+      dispatch(fechReservations());
+    }
+    if (lawyersState.status !== 'completed') {
+      dispatch(getLawyers());
+    }
+    return () => {
+      dispatch(clearFetchReservationState());
```

Comment:
After the component is unmounted, the action for reseting the state should be dispatched.

```suggestion
      dispatch(clearDeleteReservationState());
```